### PR TITLE
chore: Refactor code in ApplicationMailbox

### PR DESCRIPTION
**Pull request by jordan-brough**
No functionality changes. I was just reading through this code and thought some of these updates might make it easier to understand.

Also, a couple of minor perf improvements:

* short-circuiting as soon as we get a "true" value in some cases
* using ".exists?" instead of instantiating an ActiveRecord object
* using "match?" instead of "match"

**NOTE**: I did this in [two separate commits](https://github.com/chatwoot/chatwoot/pull/5857/commits) to make the changes easier to follow.  And also I wasn't sure if perhaps y'all would only want to use the first set of changes and not the second.

### CODE REVIEW

**Comment by jordan-brough** [comment]
@tejaswinichile any thoughts/questions on this?

File: app/mailboxes/application_mailbox.rb [hunk 1]
```diff
@@ -7,72 +7,38 @@ class ApplicationMailbox < ActionMailbox::Base
...
-    in_reply_to_match = false
-    if in_reply_to.is_a?(Array)
-      in_reply_to.each do |in_reply_to_mail|
-        in_reply_to_match ||= in_reply_to_mail.match(CONVERSATION_MESSAGE_ID_PATTERN)
-      end
-    else
-      in_reply_to_match = in_reply_to.match(CONVERSATION_MESSAGE_ID_PATTERN)
-    end
-    in_reply_to_match
+    Array.wrap(in_reply_to).any? { _1.match?(CONVERSATION_MESSAGE_ID_PATTERN) }
```
Comment (tejaswinichile) [inline app/mailboxes/application_mailbox.rb]: 💯

**Comment by tejaswinichile** [comment]
I have tested this PR on my local

1. With `:in_reply_to` UUID
2. With `:to` mail address
3. new email with new subject
4. same email with a different subject


It's working fine. Soon will add a flow to test it, and the test cases are working well.

I will check the core climate issue with the same and get it merged.

**Comment by tejaswinichile** [comment]
Screenshot 2023-01-13 at 7 34 55 PM
Screenshot 2023-01-13 at 7 34 48 PM
Screenshot 2023-01-13 at 7 35 10 PM

**Review by tejaswinichile — APPROVED** [review]
LGTM 💯 

It works on staging, @sojan-official would you check it one more time then we are good to go.

**Pull request merged**